Alameen688/MyDiary
# #159427451 Client/Postman should be able to update a user profile

#### What does this PR do?
Implements the update of user profile information
#### Description of Task to be completed?
Have the user update endpoint working
/PUT /api/v1/auth/updateuser
#### How should this be manually tested?
After cloning the repo, cd into it then
- make sure you have postgresql installed on your machine and start it.
- add a postgreSQL DATABASE_URL to your .env file
   ```
     DATABASE_URL=postgresql://username:password@localhost:5432/mydiarydb
   ```
- The RUN `npm run migrate` to create the database tables.
- Login with an existing user account via /POST api/v1/auth/login
  or register at /POST api/v1/auth/signup check [docs](http://mydiaryoxygen.herokuapp.com/api-docs/) for more info
- send a request body containing
```
{
	"fullname": "First Last",
	"email": "example@mail.com",
        "fav_quote": "Everything starts here and with you"
}
```
to /PUT /api/v1/auth/updateuser
- make sure to put the token gotten from logging in earlier in your authorization header in Postman (Client) `"Authorization" : "Bearer <the_token_here>"`
- you should get a 200 OK response after the user details of the logged in user must have been updated
with the following response body
```
{
    "status": "success",
    "data": {
       "fullname": "First Last",
	"email": "example@mail.com",
        "fav_quote": "Everything starts here and with you",
        "created_at": "2018-07-21T23:00:00.000Z",
        "updated_at": "2018-07-31T23:00:00.000Z"
    }
}
```
- if the token is not passed a 40
#### Any background context you want to provide?
none
#### What are the relevant pivotal tracker stories?
#159427451
